src/components/SignIn.js

```diff
@@ -30,17 +36,85 @@ function Copyright(props) {
 }
 
 export default function SignIn() {
-  const handleSubmit = (event) => {
+  const [formData, setFormData] = useState({
+    email: '',
+    password: '',
+  });
+
+  //update the input states when their value is changed
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData((prevData) => ({ ...prevData, [name]: value }));
+  };
+
+  const navigate = useNavigate();
+
+  // snackbar
+  const [open, setOpen] = React.useState(true);
+  const handleClose = (event, reason) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+
+    setOpen(false);
+  };
+  const action = (
+    <React.Fragment>
+      <IconButton
+        size="small"
+        aria-label="close"
+        color="inherit"
+        onClick={handleClose}
+      >
+        <CloseIcon fontSize="small" />
+      </IconButton>
+    </React.Fragment>
+  );
+  // end of snackbar
+
+  // user authentication
+  const [loginError, setLoginError] = React.useState(false);
+  let errorMessage = '';
+
+  // error handler function
+  const handleLoginError = (error) => {
+    setLoginError(true);
+    errorMessage = error.msg;
+    console.error(error);
+  };
+  const getUserFromResponse = (responseData) => {
+    return responseData.user;
+  };
+
+  // initial options needed for useLogin hook
+  const loginHookOptions = {
+    apiUrl: 'http://localhost:8000/api/v1/auth/login',
+    errorHandler: handleLoginError,
+    getUserFromResponse,
+    getJwtTokenFromResponse: false,
+  };
+
+  const {
+    submit: signIn,
+    errors,
+    loading,
+  } = useLogin(formData, loginHookOptions);
+
+  //end of user authentication
+
+  const handleSubmit = async (event) => {
     event.preventDefault();
-    const data = new FormData(event.currentTarget);
-    console.log({
-      email: data.get('email'),
-      password: data.get('password'),
-    });
+    setLoginError(false); // Reset login error state
+    const originalResponse = await signIn();
+    console.dir(originalResponse);
+    //navigate to home page only if credential is correct
+    if (originalResponse.status !== 401) {
+      navigate('/home');
+    }
```

Comment:
If the request is unsuccessful, the error will be handled by `handleLoginError`. We're not currently returning anything from there, so when there's an error `originalResponse` will be undefined. Which means this will throw an error because you can't do `undefined.status`.

Easiest thing here would probably be to check for the _success_ case and use optional chaining (`?.` to handle the null/undefined case):

```suggestion
    if (originalResponse?.status === 201) {
      navigate('/home');
    }
```

Reply:
I add that condition because even with wrong credential user was navigated to home page, I will update it with your code